### PR TITLE
Remove cruft

### PR DESCRIPTION
**Pull request by timholy**
This certainly isn't complete (e.g., the documentation will be updated separately), but it's a big chunk of #382:

- officially bump the version to 3.0.0
- require a minimum of Julia 1.10
- eliminate SnoopPrecompile
- delete a lot of version-specific cruft
- add two new extensions
- remove dependency on Requires

### CODE REVIEW

**Comment by timholy** [comment]
I don't think this needs a detailed review, but I'd appreciate whether the general plan meets approval @aviatesk @mkitti

File: NEWS.md [hunk 1]
```diff
@@ -30,3 +30,11 @@ This version implements major changes in how `parcel` works on the output of `@s
 
 Other changes:
 - A convenience utility, `timesum`, was introduced (credit to aminya)
+
+## Version 3
+
+Version 3 is a greatly slimmed repository, focusing just on those tools that are relevant for modern Julia development.
+Users of tools that were removed should stick with SnoopCompile 2.x.
+
+- The old `@snoopc` and `@snoopi` have been deleted.
```
Comment (mkitti) [inline NEWS.md:39]: Could you add a note here about what they used to do and why they have been deleted?

File: NEWS.md [hunk 2]
```diff
@@ -30,3 +30,11 @@ This version implements major changes in how `parcel` works on the output of `@s
...
+- SnoopCompileBot and SnoopPrecompile have been deleted.
```
Comment (mkitti) [inline NEWS.md:40]: I would add a note here that SnoopPrecompile is now PrecompileTools.

File: SnoopPrecompile/README.md [hunk 3]
```diff
@@ -1,5 +0,0 @@
-# SnoopPrecompile
```
Comment (mkitti) [inline SnoopPrecompile/README.md:1]: Maybe we could save this README to mainly link to PrecompileTools.jl?

File: SnoopCompileCore/src/snoopi_deep.jl [hunk 4]
```diff
@@ -1,3 +1,5 @@
+export @snoopi_deep
```
Comment (mkitti) [inline SnoopCompileCore/src/snoopi_deep.jl:1]: Were you going to alias `@snoopi_deep` as `@snoopi`?

**Review by mkitti — APPROVED** [review]
I would leave some additional breadcrumbs about the changes and the removed features.

**Review by aviatesk — APPROVED** [review]
Glad to see old code removed! I think we probably want to rename `@snoopi_deep` to `@snoopi`, while also deprecating `@snoopi_deep` and making it an alias for the new `@snoopi`?

**Comment by timholy** [comment]
How about a more wholesale renaming?
- `@snoopi_deep` -> `@snoop_infer`
- `@snoopl` -> `@snoop_llvm`
- `@snoopr` -> `@snoop_inval`

**Comment by aviatesk** [comment]
> How about a more wholesale renaming?
> 
> * `@snoopi_deep` -> `@snoop_infer`
> * `@snoopl` -> `@snoop_llvm`
> * `@snoopr` -> `@snoop_inval`

That sounds good to me. I'm all for that.

File: SnoopCompileCore/src/snoopi_deep.jl (hunk 4 above)
Comment (timholy) [reply SnoopCompileCore/src/snoopi_deep.jl:1]: In a separate PR, yes. Let's keep this one focused on deleting stuff.

**Comment by timholy** [comment]
Further changes, like renaming the main data-collection macros, are things I intend for a future PR.

**Pull request merged**